### DISCUSSION TITLE
key-auth plugin support customize the error msg

### DISCUSSION THREAD

**Discussion by nimohunter** [apache/apisix#6025] [Ideas]
when i use the key-auth, i found we can customize keyname apikey to any word i like to.

but the error msg always return "{"message" = "Missing related consumer"}" and code is 401.

can we customize the error msg?

**Comment by juzhiyuan** [comment]
It's more like a proposal to support more detailed error messages 😄

**Reply by nimohunter** [reply]
how about we can customize the error code and error msg in key-auth config? 
the default is 401 with "Missing API key found in request" , "Missing related consumer" ," Invalid API key in request"

**Reply by juzhiyuan** [reply]
Ya, this will be more semantic, what do you think? @tzssangglass @leslie-tsang

**Reply by leslie-tsang** [reply]
Of course, we can add this feature to the `key-auth`, @nimohunter Would you like to have a try ?

**Reply by nimohunter** [reply]
OK ,i'll try to do it.

**Reply by leslie-tsang** [reply]
If you need any help, please let me know, here are some example[1][2] can do help.
[1] https://github.com/apache/apisix/pull/5732
[2] https://github.com/apache/apisix/pull/4849

**Reply by nimohunter** [reply]
hi, here is the mr : #6040

**Reply by Routhinator** [reply]
Is there a way to add a default backend and have it return a 401 page instead of this?
